homeassistant/components/modern_forms/switch.py

```diff
@@ -0,0 +1,145 @@
+    ) -> None:
+        """Initialize Modern Forms Away mode switch."""
+        super().__init__(
+            coordinator=coordinator,
+            entry_id=entry_id,
+            icon="mdi:airplane-takeoff",
+            key="away_mode",
+            name=f"{coordinator.data.info.device_name} Away Mode",
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return bool(self.coordinator.data.state.away_mode_enabled)
+
+    @modernforms_exception_handler
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the Modern Forms Away mode switch."""
+        await self.coordinator.modern_forms.away(away=False)
+
+    @modernforms_exception_handler
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the Modern Forms Away mode switch."""
+        await self.coordinator.modern_forms.away(away=True)
+
+
+class ModernFormsAdaptiveLearningSwitch(ModernFormsSwitch):
+    """Defines a Modern Forms Adaptive Learning switch."""
+
+    def __init__(
+        self, entry_id: str, coordinator: ModernFormsDataUpdateCoordinator
+    ) -> None:
+        """Initialize Modern Forms Adaptive Learning switch."""
+        super().__init__(
+            coordinator=coordinator,
+            entry_id=entry_id,
+            icon="mdi:school-outline",
+            key="adaptive_learning",
+            name=f"{coordinator.data.info.device_name} Adaptive Learning",
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return bool(self.coordinator.data.state.adaptive_learning_enabled)
+
+    @modernforms_exception_handler
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the Modern Forms Adaptive Learning switch."""
+        await self.coordinator.modern_forms.adaptive_learning(adaptive_learning=False)
+
+    @modernforms_exception_handler
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the Modern Forms Adaptive Learning switch."""
+        await self.coordinator.modern_forms.adaptive_learning(adaptive_learning=True)
+
+
+class ModernFormsRebootSwitch(ModernFormsSwitch):
+    """Defines a Modern Forms Reboot switch."""
+
+    def __init__(
+        self, entry_id: str, coordinator: ModernFormsDataUpdateCoordinator
+    ) -> None:
+        """Initialize Modern Forms Reboot switch."""
+        super().__init__(
+            coordinator=coordinator,
+            entry_id=entry_id,
+            icon="mdi:restart",
+            key="reboot",
+            name=f"{coordinator.data.info.device_name} Reboot",
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return false as the switch will never be on."""
+        return False
+
+    @modernforms_exception_handler
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the Modern Forms reboot switch."""
+        pass  # pylint: disable=unnecessary-pass
```

Comment:
The ignore is there because the default behaviour for async_turn_off is to raise a NotImplemented exception.  In this case we are implemented just don't want it doing anything.  The reboot will only react to a turn on and immediate kills the control unit so it doesn't get a response to the on and reverts to off in a few seconds anyways.

Reply:
We expect a switch to have two states, on and off which can be set and read.
I don't think `switch` is a right abstraction for the reboot functionality, it would be better modelled as an integration level service.

Reply:
yeah think your right.  removed reboot switch